IDinsight/aaq-core
# Multi-user retrieval part 1: Add per-user keys and filter content before responding

Reviewer: @sidravi1 
Estimate: 30mins

---

## Ticket

Fixes: [477](https://idinsight.atlassian.net/browse/AAQ-477?atlOrigin=eyJpIjoiY2NmZTlkNWU1OTkzNDBiODkyMDdhYWYyOTA1YjhhOGMiLCJwIjoiaiJ9)

## Description

### Goal

- Give each user a unique API key which is used to only use their contents when fetching contents.

### Changes

- Give each user a retrieval key (new column in user table)
- Filter contents in QA endpoints
- Check for user_id in feedback endpoints. 
  - Q: **Should we add a user_id column to the feedback tables? Probably not, right?**
- Update deployment
- Fix pytests
- Add user_id to query table now that we know who the requests are associated with
  - **Should we add a user_id column to the query-response and error table? Probably not, right?**

### Future Tasks

- SECURITY: Hash the API keys instead of saving them in the DB as strings!
- Make new tests

## How has this been tested?

- run tests
- run dev setup and check functionality
- run deployment (with fresh `template.env` copy and removed volumes and images) and check

## To-do before merge

- Merge #175 

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant
